assets/src/hooks/useDetour.ts

```diff
@@ -2,29 +2,37 @@ import { useEffect, useMemo, useState } from "react"
 import { ShapePoint } from "../schedule"
 import { fetchDetourDirections, fetchFinishedDetour } from "../api"
 import { DetourShape, FinishedDetour, OriginalRoute } from "../models/detour"
-import { isOk } from "../util/fetchResult"
-
-const useDetourDirections = (shapePoints: ShapePoint[]) => {
-  const [detourShape, setDetourShape] = useState<ShapePoint[]>([])
-  const [directions, setDirections] = useState<
-    DetourShape["directions"] | undefined
-  >(undefined)
+import {
+  ok,
+  FetchResult,
+  isOk,
+  isFetchError,
+  fetchError,
+  loading,
+} from "../util/fetchResult"
+
+const useDetourDirections = (
+  shapePoints: ShapePoint[]
+): FetchResult<DetourShape> => {
+  const [detourShape, setDetourShape] = useState<FetchResult<DetourShape>>(
+    loading()
+  )
 
   useEffect(() => {
     let shouldUpdate = true
 
     if (shapePoints.length < 2) {
       // We expect not to have any directions or shape if we don't have at
       // least two points to route between
-      setDetourShape([])
-      setDirections(undefined)
+      setDetourShape(ok({ coordinates: [], directions: undefined }))
       return
     }
 
     fetchDetourDirections(shapePoints).then((detourInfo) => {
       if (isOk(detourInfo) && shouldUpdate) {
-        setDetourShape(detourInfo.ok.coordinates)
-        setDirections(detourInfo.ok.directions)
+        setDetourShape(ok(detourInfo.ok))
```

Comment:
Can we just replace this whole if statement with 👇?
```jsx
setDetourShape(detourInfo)
```

It looks like right now we're just destructuring the two possibilities and then restructuring them back to exactly what they were before.

(I can't do it as a suggestion because you didn't modify line 32, so Github won't let me include that in the diff ☹️)

Reply:
Ah good call, see 1b5fa7a8!